icu4c/source/i18n/unitconverter.cpp

```diff
@@ -5,14 +5,494 @@
 
 #if !UCONFIG_NO_FORMATTING
 
+#include <cmath>
+
+#include "charstr.h"
+#include "double-conversion.h"
+#include "measunit_impl.h"
+#include "resource.h"
+#include "uassert.h"
+#include "unicode/stringpiece.h"
+#include "unicode/unistr.h"
+#include "unicode/utypes.h"
+#include "unitconverter.h"
+#include "uresimp.h"
+
 U_NAMESPACE_BEGIN
 
+namespace {
+
+using icu::double_conversion::StringToDoubleConverter;
+
+/* Internal Structure */
+
+// Represents a raw convert unit.
+struct ConvertUnit {
+    StringPiece source;
+    StringPiece target;
```

Comment:
Good point.  I would change the type when the data loading logic is implemented, when we know more about exactly what we need to do.

Reply:
I think StringPiece is better because of two things:
  1 - it is cheap to pass around and to have multiple substrings.
  2 - we do not need to keep the data, we just want to use them until we get the conversion rate.
  3 - we can use CharString if we want to save the data for future use.